### PR TITLE
fix detecting config changes. Closes #311

### LINKED ISSUE #311: Config always detected as changed
At least that's what I'm seeing now. We might be better off just json-marshalling that thing and doing a string comparison

### PR DESCRIPTION
**Pull request by mthenw**
@tj not sure if this is what you mean but it works :)

### CODE REVIEW

File: function/function.go [hunk 1]
```diff
@@ -576,39 +587,40 @@ func (f *Function) currentVersionAlias() (*lambda.AliasConfiguration, error) {
...
-	}
+	remoteConfig, _ := json.Marshal(diffConfig{
+		Description: *config.Configuration.Description,
+		Memory:      *config.Configuration.MemorySize,
+		Timeout:     *config.Configuration.Timeout,
+		Role:        *config.Configuration.Role,
+		Handler:     *config.Configuration.Handler,
+		VPC: vpc.VPC{
+			Subnets:        aws.StringValueSlice(config.Configuration.VpcConfig.SubnetIds),
+			SecurityGroups: aws.StringValueSlice(config.Configuration.VpcConfig.SecurityGroupIds),
```
Comment (tj) [inline function/function.go]: the rest looks good to me but what about the case when `config.Configuration.VpcConfig` == nil

Comment (mthenw) [inline function/function.go]: won't be. it's initialized with empty arrays.

**Comment by mthenw** [comment]
I'm merging as the rest is fine :)

File: function/function.go (hunk 1 above)
Comment (tj) [inline function/function.go]: I don't think it's initialized on the AWS end properly, I was having errors but I'll install master's HEAD and check in a bit

Comment (mthenw) [inline function/function.go]: yeah, please check. based on my tests it's initialized.

Comment (tj) [inline function/function.go]: weiiird seems fine, not sure what I hit

**Pull request merged**